### ISSUE TITLE
gradient back propagation in tf.contrib.image.transform

### ISSUE THREAD

**Issue by xuyingxiao** [tensorflow/tensorflow#30413]
**System information** Ubuntu 16.04
TensorFlow version 1.10.1

tf.contrib.image.transform applies the given transform(s) to the image(s). But gradients are not backpropagated into transformation parameters https://www.tensorflow.org/api_docs/python/tf/contrib/image/transform. 
Affine transformations can be constructed using a series of translations, scales, rotations and shears. It could calculate the gradient and back propagation, at least in rotation,scale and translations. I noticed that tf.contrib.image.rotate and tf.contrib.image.translate do not emphasize that gradients cannot propagate backwards.
I'd like to know how to back propagation into the transformation parameters especially in affine transform. Do I need to explicitly define the gradient using gradient_map_override?（#17442）

Any suggestions would be helpful. Thanks

This may change the current api.

**Comment by rmothukuru** [comment]
@xuyingxiao ,
Can you please try using latest version of Tensorflow (1.14) and confirm if the issue still persists.

If the issue persists, In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by xuyingxiao** [comment]
Yeah,there is a demo https://github.com/xuyingxiao/image_transform
The main loss function lines:
    `h10 = tf.cos(pred_rotate_theta)*pred_scale_x
    h11 = tf.sin(pred_rotate_theta)*pred_scale_x
    h12 = pred_offset_x
    h20 = -tf.sin(pred_rotate_theta)*pred_scale_x
    h21 = tf.cos(pred_rotate_theta)*pred_scale_y
    h22 = pred_offset_y
    h30 = tf.constant([0.0])
    h31 = tf.constant([0.0])
    h32 = tf.constant([1.0])

    matrix_3_3 = tf.linalg.inv(tf.reshape(tf.concat([h10,h11,h12,h20,h21,h22,h30,h31,h32],axis=-1),shape=(3,3)))
    matrix_1_8 = tf.contrib.image.matrices_to_flat_transforms(matrix_3_3)

    warp_image = tf.contrib.image.transform(input_disp_polygon_map,matrix_1_8,interpolation="BILINEAR",name=None)
    print("warp_image",warp_image.shape)
	warp_loss = tf.reduce_mean(tf.abs(warp_image - input_gt_polygon_map))`

**Commit line 504 runs with only warp_loss, THANKS!** 
Any suggestions would be helpful.

**Comment by nhatuan84** [comment]
did someone solve this?

**Comment by nhatuan84** [comment]
@xuyingxiao 

did someone solve this?

**Comment by xuyingxiao** [comment]
> @xuyingxiao
> 
> did someone solve this?

Not yet.

**Comment by nhatuan84** [comment]
This solved problem: https://github.com/kevinzakka/spatial-transformer-network
Enjoy :)

**Comment by ymodak** [comment]
`contrib` module is deprecated with TF 2.X 
You may still use subset if features by using `tf addons` module. Thanks!
https://www.tensorflow.org/addons/api_docs/python/tfa/image
